### PR TITLE
Enable CA1822: Mark private members as static

### PR DESCRIPTION
**Pull request by xtqqczze**
https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822

Non-breaking change as only enabled for non-public API surface.

### CODE REVIEW

**Comment by iSazonov** [comment]
@xtqqczze Please split the PR.

**Comment by xtqqczze** [comment]
> @xtqqczze Please split the PR.

Split is completed.

**Comment by xtqqczze** [comment]
@iSazonov Please could you review.

File: .globalconfig [hunk 1]
```diff
@@ -280,7 +280,8 @@ dotnet_diagnostic.CA1820.severity = none
 dotnet_diagnostic.CA1821.severity = warning
 
 # CA1822: Mark members as static
-dotnet_diagnostic.CA1822.severity = suggestion
+dotnet_diagnostic.CA1822.severity = warning
+dotnet_code_quality.ca1822.api_surface = private
```
Comment (iSazonov) [inline .globalconfig:284]: What about internal?

Comment (xtqqczze) [reply .globalconfig:284]: For new PR. I've updated the title.

**Review by iSazonov — APPROVED** [review]

**Comment by xtqqczze** [comment]
@iSazonov This is blocking #14335, can we merge?

File: src/TypeCatalogGen/TypeCatalogGen.cs [hunk 2]
```diff
@@ -431,7 +431,7 @@ namespace System.Management.Automation
 {{
     internal partial class PowerShellAssemblyLoadContext
     {{
-        private Dictionary<string, string> InitializeTypeCatalog()
+        private static Dictionary<string, string> InitializeTypeCatalog()
         {{
```
Comment (iSazonov) [inline src/TypeCatalogGen/TypeCatalogGen.cs:435]: @rjmholt Can you approve the change?

**Review by rjmholt — APPROVED** [review]

**Pull request merged**